src/SprykerEco/Zed/Inxmail/Business/Mapper/Order/AbstractOrderMapper.php

```diff
@@ -0,0 +1,308 @@
+use SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToMoneyFacadeBridgeInterface;
+use SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToProductFacadeBridgeInterface;
+use SprykerEco\Zed\Inxmail\InxmailConfig;
+
+abstract class AbstractOrderMapper implements OrderMapperInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Inxmail\InxmailConfig
+     */
+    protected $config;
+
+    /**
+     * @var \Spryker\Service\UtilDateTime\UtilDateTimeServiceInterface
+     */
+    protected $dateTimeService;
+
+    /**
+     * @var \SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToMoneyFacadeBridgeInterface
+     */
+    protected $moneyFacade;
+
+    /**
+     * @var \SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToProductFacadeBridgeInterface
+     */
+    protected $productFacade;
+
+    /**
+     * @var \SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToLocaleFacadeInterface
+     */
+    protected $localeFacade;
+
+    /**
+     * @param \SprykerEco\Zed\Inxmail\InxmailConfig $config
+     * @param \Spryker\Service\UtilDateTime\UtilDateTimeServiceInterface $dateTimeService
+     * @param \SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToMoneyFacadeBridgeInterface $moneyFacade
+     * @param \SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToProductFacadeBridgeInterface $productFacade
+     * @param \SprykerEco\Zed\Inxmail\Dependency\Facade\InxmailToLocaleFacadeInterface $localeFacade
+     */
+    public function __construct(
+        InxmailConfig $config,
+        UtilDateTimeServiceInterface $dateTimeService,
+        InxmailToMoneyFacadeBridgeInterface $moneyFacade,
+        InxmailToProductFacadeBridgeInterface $productFacade,
+        InxmailToLocaleFacadeInterface $localeFacade
+    ) {
+        $this->config = $config;
+        $this->dateTimeService = $dateTimeService;
+        $this->moneyFacade = $moneyFacade;
+        $this->productFacade = $productFacade;
+        $this->localeFacade = $localeFacade;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return \Generated\Shared\Transfer\InxmailRequestTransfer
+     */
+    public function map(OrderTransfer $orderTransfer): InxmailRequestTransfer
+    {
+        $inxmailRequestTransfer = new InxmailRequestTransfer();
+        $inxmailRequestTransfer->setEvent($this->getEvent());
+        $inxmailRequestTransfer->setTransactionId($orderTransfer->getOrderReference());
+        $inxmailRequestTransfer->setPayload($this->getPayload($orderTransfer));
+
+        return $inxmailRequestTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\OrderTransfer $orderTransfer
+     *
+     * @return array
+     */
+    protected function getPayload(OrderTransfer $orderTransfer): array
+    {
+        $locale = $orderTransfer->getCustomer()->getLocale() ?
+            $orderTransfer->getCustomer()->getLocale()->getLocaleName() :
+            $this->localeFacade->getCurrentLocaleName();
+
+        $payload = [
+            'Customer' => [
```

Comment:
Same. Do you see? If something changes you'll have to fix it in many places.

Reply:
Move ALL this strings to constants.

Reply:
It's okay for mappers.